perun-core/src/main/java/cz/metacentrum/perun/core/impl/AuthzResolverImpl.java

```diff
@@ -128,6 +128,28 @@ public AuthzRoles getRoles(User user) throws InternalErrorException {
 		return authzRoles;
 	}
 
+	public AuthzRoles getRoles(Group group) throws InternalErrorException {
+		AuthzRoles authzRoles = new AuthzRoles();
+
+		if (group != null) {
+			try {
+				// Get roles from Authz table
+				List<Pair<Role, Map<String, Set<Integer>>>> authzRolesPairs = jdbc.query("select " + authzRoleMappingSelectQuery
+				+ ", distinct roles.name as role_name from authz left join roles on authz.role_id=roles.id where authz.authorized_group_id=?",
```

Comment:
I believe distinct in the middle of select is not OK, since you would need to agregate rest of columns. Try to remove it. Uniqueness of result is ensured by upper layer, where your call and return ``getRolesNames()`` on the ``AuthzRoles`` object returned here (result is a set of names), so "distinct" shouldn't be needed.
```
cz.metacentrum.perun.core.api.exceptions.InternalErrorException: Error 163d48d72c2: PreparedStatementCallback; bad SQL grammar [select  authz.user_id as authz_user_id, authz.role_id as authz_
role_id,authz.authorized_group_id as authz_authorized_group_id, authz.vo_id as pb_vo_id, authz.group_id as pb_group_id, authz.facility_id as pb_facility_id, authz.member_id as pb_member_id, a
uthz.resource_id as pb_resource_id, authz.service_id as pb_service_id, authz.service_principal_id as pb_user_id, authz.security_team_id as pb_security_team_id, authz.sponsored_user_id as pb_s
ponsored_user_id, distinct roles.name as role_name from authz left join roles on authz.role_id=roles.id where authz.authorized_group_id=?]; nested exception is org.postgresql.util.PSQLExcepti
on: ERROR: syntax error at or near "distinct"
  Pozice: 464
```